### PR TITLE
feat: enable ECS task monitoring

### PR DESCRIPTION
**Pull request by michaelwittig**
@andreaswittig what do you think? it is partly covered by the ECS service monitoring. But might be still important to detect issues early.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1445,6 +1445,42 @@ resource "aws_cloudwatch_event_target" "ec2_spot_instance_interruption" {
...
+    "aws.ecs"
+  ],
+  "detail-type": [
+    "ECS Task State Change"
+  ],
+  "detail": {
+    "lastStatus": ["STOPPED"],
+    "stopCode": [
+      "TaskFailedToStart",
+      "EssentialContainerExited"
```
Comment (andreaswittig) [inline main.tf]: Isn't it possible that a container exits in a well-planned way?

Comment (michaelwittig) [reply main.tf]: You are right. We should check the exit code <> 0

File: main.tf [hunk 2]
```diff
@@ -1548,43 +1619,6 @@ resource "aws_cloudwatch_event_target" "ecs_deployment_notifications" {
 
 
 
-resource "aws_cloudwatch_event_rule" "ecs_spot_placement_failure" {
```
Comment (michaelwittig) [inline main.tf:1551]: Rule is a subset of the ecs_service_failed rule

File: main.tf [hunk 3]
```diff
@@ -1602,6 +1636,7 @@ resource "aws_cloudwatch_event_rule" "ecs_spot_interruption" {
   ],
   "detail": {
     "stopCode": [
+      "SpotInterruption",
```
Comment (michaelwittig) [inline main.tf:1639]: Do we want interruptions to be forwarded for tasks running in a service? The service takes care of this notice by trying to launch new capacity.

One similarity: We show those spot interruptions for EC2 instances no matter if they run in an ASG with Capacity Rebalancing or not.

**Comment by michaelwittig** [comment]
@andreaswittig please check again

File: main.tf (hunk 3 above)
Comment (andreaswittig) [reply main.tf:1639]: Good question. Maybe we should stick to what we have for ASG?

**Review by andreaswittig — APPROVED** [review]

**Pull request merged**